### PR TITLE
Mech id upload buttons

### PR DESCRIPTION
**Pull request by SonixApache**
savesalottatime

🆑 
 - rscadd: Mech ID upload panels now come with [add all] and [delete all] options

### CODE REVIEW

**Comment by tuypo1** [comment]
Yes fucking please
It wouldn't be so bad if clicking add did not reset the window so you have to put it full screen to be able to do it with decent speed when you want to grant aa

**Comment by Kharkov72** [comment]
holy fucking based

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -1892,12 +1897,25 @@
 		operation_req_access += topic_filter.getNum("add_req_access")
 		output_access_dialog(topic_filter.getObj("id_card"),topic_filter.getMob("user"))
 		return
+	if(href_list["add_all_req_access"] && add_req_access && topic_filter.getObj("id_card"))
+		if(!in_range(src, usr))
+			return
+		for(var/a in topic_filter.getObj("id_card").access)
```
Comment (Kurfursten) [inline code/game/mecha/mecha.dm]: iirc this works the same as a trust operator, so you should separate it out into different lines

Comment (SonixApache) [reply code/game/mecha/mecha.dm]: I'm not sure what that means or how I would do as you ask.
```
var/card = getobj,
var/accessed = card.access
for var/a in accesses
```
?

Comment (Kurfursten) [reply code/game/mecha/mecha.dm]: Yes, that's fine

**Pull request merged**